commit c7ca9757bda35ff9ce27ab42f2cb8b84d983e6ad
Author: Hannes Reinecke <hare@suse.de>
Date:   Thu Nov 16 13:14:35 2023 +0100

    nvme: blank out authentication fabrics options if not configured
    
    If the config option NVME_HOST_AUTH is not selected we should not
    accept the corresponding fabrics options. This allows userspace
    to detect if NVMe authentication has been enabled for the kernel.
    
    Cc: Shin'ichiro Kawasaki <shinichiro.kawasaki@wdc.com>
    Fixes: f50fff73d620 ("nvme: implement In-Band authentication")
    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Tested-by: Shin'ichiro Kawasaki <shinichiro.kawasaki@wdc.com>
    Reviewed-by: Daniel Wagner <dwagner@suse.de>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Keith Busch <kbusch@kernel.org>

diff --git a/drivers/nvme/host/fabrics.c b/drivers/nvme/host/fabrics.c
index 4673ead69c5f..aa88606a44c4 100644
--- a/drivers/nvme/host/fabrics.c
+++ b/drivers/nvme/host/fabrics.c
@@ -667,8 +667,10 @@ static const match_table_t opt_tokens = {
 #endif
 	{ NVMF_OPT_FAIL_FAST_TMO,	"fast_io_fail_tmo=%d"	},
 	{ NVMF_OPT_DISCOVERY,		"discovery"		},
+#ifdef CONFIG_NVME_HOST_AUTH
 	{ NVMF_OPT_DHCHAP_SECRET,	"dhchap_secret=%s"	},
 	{ NVMF_OPT_DHCHAP_CTRL_SECRET,	"dhchap_ctrl_secret=%s"	},
+#endif
 #ifdef CONFIG_NVME_TCP_TLS
 	{ NVMF_OPT_TLS,			"tls"			},
 #endif